parser-typechecker/src/Unison/Codebase/Causal.hs

```diff
@@ -132,27 +164,38 @@ instance Hashable (RawHash h) where
 -- Find the lowest common ancestor of two causals.
 lca :: Monad m => Causal m h e -> Causal m h e -> m (Maybe (Causal m h e))
 lca a b =
-  go Set.empty Set.empty (Seq.singleton $ pure a) . Seq.singleton $ pure b
- where
+  lca' (Seq.singleton $ pure a) . Seq.singleton $ pure b
+
+-- Find the lowest common ancestor of two causal-forests.
+-- The convention is that the causals in a forest are the ancestors of some
+-- other Causal.
+lca'
+  :: Monad m
+  => Seq (m (Causal m h e))
+  -> Seq (m (Causal m h e))
+  -> m (Maybe (Causal m h e))
+lca' = go Set.empty Set.empty where
```

Comment:
What's the importance of the causal forest? Would you explain here or in the PR comments